File: app/src/main/java/com/minew/mst03demo/AuthApiService.java
package com.minew.mst03demo;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface AuthApiService {
    
    @POST("auth/initLogin")
    Call<AuthModels.InitLoginResponse> initLogin(@Body AuthModels.InitLoginRequest request);
    
    @POST("auth/login")
    Call<AuthModels.LoginResponse> login(@Body AuthModels.LoginRequest request);
    
    @POST("auth/forgotPassword")
    Call<AuthModels.ForgotPasswordResponse> forgotPassword(@Body AuthModels.ForgotPasswordRequest request);
    
    @POST("auth/verifyOtp")
    Call<AuthModels.VerifyOtpResponse> verifyOtp(@Body AuthModels.VerifyOtpRequest request);
    
    @POST("auth/initPassword")
    Call<AuthModels.InitPasswordResponse> initPassword(@Body AuthModels.InitPasswordRequest request);
} 

File: app/src/main/java/com/choruscoldchain/auth/AuthApiService.kt
package com.choruscoldchain.auth

import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.Header
import retrofit2.http.POST

interface AuthApiService {
    @POST("auth/initLogin")
    fun initLogin(@Body request: AuthModels.InitLoginRequest): Call<AuthModels.InitLoginResponse>

    @POST("auth/login")
    fun login(@Body request: AuthModels.LoginRequest): Call<AuthModels.LoginResponse>

    @POST("auth/forgotPassword")
    fun forgotPassword(@Body request: AuthModels.ForgotPasswordRequest): Call<AuthModels.ForgotPasswordResponse>

    @POST("auth/verifyOtp")
    fun verifyOtp(@Body request: AuthModels.VerifyOtpRequest): Call<AuthModels.VerifyOtpResponse>

    @POST("auth/initPassword")
    fun initPassword(@Body request: AuthModels.InitPasswordRequest): Call<AuthModels.InitPasswordResponse>

    @POST("auth/resetPassword")
    fun resetPassword(
        @Header("Authorization") authorization: String,
        @Body request: AuthModels.ResetPasswordRequest
    ): Call<AuthModels.ResetPasswordResponse>
}



